### PR TITLE
feat: pipeline display task param detail

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
pipeline display task param detail

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=318516&iterationID=1280&pId=317574&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    pipeline display task param detail           |
| 🇨🇳 中文    |      流水线详情展示任务详细参数        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chengjoey** [comment]
image

File: apistructs/pipeline_task.go [hunk 1]
```diff
@@ -31,6 +31,19 @@ const (
...
+
+const (
+	DefaultTaskParamSource TaskParamSource = "default"
+	UserTaskParamSource    TaskParamSource = "user"
+	MergedTaskParamSource  TaskParamSource = "merged"
+)
+
+type TaskParamDetail struct {
+	Name  string                     `json:"name"`
+	Value map[TaskParamSource]string `json:"value"`
```
Comment (sfwn) [inline apistructs/pipeline_task.go:44]: Values

File: internal/tools/pipeline/services/pipelinesvc/detail.go [hunk 2]
```diff
@@ -130,12 +132,31 @@ func (s *PipelineSvc) Detail(pipelineID uint64) (*apistructs.PipelineDetailDTO,
 			if p.Status.IsStopByUser() && task.Status == apistructs.PipelineStatusAnalyzed {
 				task.Status = apistructs.PipelineStatusNoNeedBySystem
 			}
-			taskDTOs = append(taskDTOs, *task.Convert2DTO())
+			taskDTO := *task.Convert2DTO()
+			if action, ok := actionMap[task.Type]; ok {
```
Comment (sfwn) [inline internal/tools/pipeline/services/pipelinesvc/detail.go:136]: task should always have params, regardless action definition exist or not.

**Review by sfwn — CHANGES_REQUESTED** [review]
add an screenshot if have

File: apistructs/pipeline_task.go (hunk 1 above)
Comment (chengjoey) [reply apistructs/pipeline_task.go]: adjusted

File: internal/tools/pipeline/services/pipelinesvc/detail.go (hunk 2 above)
Comment (chengjoey) [reply internal/tools/pipeline/services/pipelinesvc/detail.go:138]: adjusted

**Review by sfwn — APPROVED** [review]

**Pull request merged**